compiler-plugin/src/main/kotlin/com/google/devtools/ksp/KotlinSymbolProcessingExtension.kt

```diff
@@ -128,16 +135,23 @@ abstract class AbstractKotlinSymbolProcessingExtension(val options: KspOptions,
                     anyChangesWildcard,
                     ksFiles
             )
-            processors.forEach {
-                it.init(options.processingOptions, KotlinVersion.CURRENT, codeGenerator, logger)
-                deferredSymbols[it] = mutableListOf()
+            processors.forEach { processor ->
+                handleException {
+                    processor.init(options.processingOptions, KotlinVersion.CURRENT, codeGenerator, logger)
+                }?.let { return it }
+                deferredSymbols[processor] = mutableListOf()
             }
             initialized = true
         }
-        processors.forEach {
-            deferredSymbols[it] = it.process(resolver)
-            if (!deferredSymbols.containsKey(it) || deferredSymbols[it]!!.isEmpty()) {
-                deferredSymbols.remove(it)
+        processors.forEach processing@{ processor ->
```

Comment:
Should we check `logger.hasError()` here, in case some processor fail in `init()`?